server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessHealth.java

```diff
@@ -37,20 +44,55 @@ public class ClusterAwarenessHealth implements Writeable {
-            displayUnassignedShardLevelInfo,
-            clusterState
-        );
+        boolean displayUnassignedShardLevelInfo;
+        ClusterAwarenessAttributesHealth clusterAwarenessAttributesHealth;
+        clusterAwarenessAttributesHealthMap = new HashMap<>();
+        // Helper function to check if we need health for all or for one awareness attribute.
+        boolean displayAllAwarenessAttribute = shouldDisplayAllAwarenessAttributeHealth(awarenessAttributeName);
+        if (!displayAllAwarenessAttribute) {
+            displayUnassignedShardLevelInfo = canCalcUnassignedShards(clusterSettings, awarenessAttributeName);
+            clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(
+                awarenessAttributeName,
+                displayUnassignedShardLevelInfo,
+                clusterState
+            );
+            clusterAwarenessAttributesHealthMap.put(awarenessAttributeName, clusterAwarenessAttributesHealth);
+        } else {
+            List<String> awarenessAttributeList = clusterSettings.get(
+                AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING
+            );
+            for (String awarenessAttribute : awarenessAttributeList) {
+                displayUnassignedShardLevelInfo = canCalcUnassignedShards(clusterSettings, awarenessAttributeName);
+                clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(
+                    awarenessAttribute,
+                    displayUnassignedShardLevelInfo,
+                    clusterState
+                );
+                clusterAwarenessAttributesHealthMap.put(awarenessAttribute, clusterAwarenessAttributesHealth);
+            }
+        }
     }
 
     public ClusterAwarenessHealth(final StreamInput in) throws IOException {
-        clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(in);
+        int size = in.readVInt();
+        if (size > 0) {
+            clusterAwarenessAttributesHealthMap = new HashMap<>(size);
+            for (int i = 0; i < size; i++) {
+                ClusterAwarenessAttributesHealth clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(in);
+                clusterAwarenessAttributesHealthMap.put(
+                    clusterAwarenessAttributesHealth.getAwarenessAttributeName(),
+                    clusterAwarenessAttributesHealth
+                );
+            }
+        } else {
+            clusterAwarenessAttributesHealthMap = Collections.emptyMap();
+        }
     }
 
-    public ClusterAwarenessHealth(String awarenessAttribute) {
-        this.clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(awarenessAttribute, Collections.emptyMap());
+    private boolean shouldDisplayAllAwarenessAttributeHealth(String awarenessAttribute) {
+        if (awarenessAttribute == null) {
+            return true;
+        } else return awarenessAttribute.isBlank();
     }
```

Comment:
lets make it inline and use below may be . 

```
(awarenessAttribute == null  || awarenessAttribute.isBlank());
```

Reply:
This will cause null pointer exception if we move in one line so bifurcated it.